test/core/security/local_security_connector_test.cc

```diff
@@ -0,0 +1,139 @@
+
+const grpc_endpoint_vtable kUnixEndpointVtable = {
+    nullptr, nullptr, nullptr, nullptr,
+    nullptr, nullptr, nullptr, GetLocalUnixAddress,
+    nullptr, nullptr};
+
+absl::string_view GetLocalTcpAddress(grpc_endpoint* /*ep*/) {
+  return "ipv4:127.0.0.1:12667";
+}
+
+const grpc_endpoint_vtable kTcpEndpointVtable = {
+    nullptr, nullptr, nullptr, nullptr,
+    nullptr, nullptr, nullptr, GetLocalTcpAddress,
+    nullptr, nullptr};
+
+void CheckSecurityLevelForServer(grpc_local_connect_type connect_type,
+                                 tsi_security_level level, grpc_endpoint ep) {
+  grpc_server_credentials* server_creds =
+      grpc_local_server_credentials_create(connect_type);
+  ChannelArgs args;
+  RefCountedPtr<grpc_server_security_connector> connector =
+      server_creds->create_security_connector(args);
+  ASSERT_NE(connector, nullptr);
+  tsi_peer peer;
+  CHECK(tsi_construct_peer(0, &peer) == TSI_OK);
+
+  RefCountedPtr<grpc_auth_context> auth_context;
+  connector->check_peer(peer, &ep, args, &auth_context, nullptr);
+  tsi_peer_destruct(&peer);
+  auto it = grpc_auth_context_find_properties_by_name(
+      auth_context.get(), GRPC_TRANSPORT_SECURITY_LEVEL_PROPERTY_NAME);
+  const grpc_auth_property* prop = grpc_auth_property_iterator_next(&it);
+  ASSERT_NE(prop, nullptr);
+  EXPECT_STREQ(prop->value, tsi_security_level_to_string(level));
+  connector.reset();
+  auth_context.reset();
+  grpc_server_credentials_release(server_creds);
+}
+
+void CheckSecurityLevelForChannel(grpc_local_connect_type connect_type,
+                                  tsi_security_level level, grpc_endpoint ep) {
+  grpc_channel_credentials* channel_creds =
+      grpc_local_credentials_create(connect_type);
+  ChannelArgs args;
+  args = args.Set((char*)GRPC_ARG_SERVER_URI, (char*)"unix:");
+  RefCountedPtr<grpc_channel_security_connector> connector =
+      channel_creds->create_security_connector(nullptr, "unix:", &args);
+  ASSERT_NE(connector, nullptr);
+  tsi_peer peer;
+  CHECK(tsi_construct_peer(0, &peer) == TSI_OK);
+  RefCountedPtr<grpc_auth_context> auth_context;
+  connector->check_peer(peer, &ep, args, &auth_context, nullptr);
+  tsi_peer_destruct(&peer);
+
+  auto it = grpc_auth_context_find_properties_by_name(
+      auth_context.get(), GRPC_TRANSPORT_SECURITY_LEVEL_PROPERTY_NAME);
+  const grpc_auth_property* prop = grpc_auth_property_iterator_next(&it);
+  ASSERT_NE(prop, nullptr);
+  EXPECT_STREQ(prop->value, tsi_security_level_to_string(level));
+
+  connector.reset();
+  auth_context.reset();
+  grpc_channel_credentials_release(channel_creds);
+}
+
+TEST(LocalSecurityConnectorTest, CheckSecurityLevelOfUdsConnectionServer) {
+  grpc_endpoint ep;
+  ep.vtable = &kUnixEndpointVtable;
+  CheckSecurityLevelForServer(UDS, TSI_PRIVACY_AND_INTEGRITY, ep);
+}
+
+TEST(LocalSecurityConnectorTest, SecurityLevelOfTcpConnectionServer) {
+  if (!IsLocalConnectorSecureEnabled()) {
+    return;
```

Comment:
Instead of skipping the entire test when the experiment is not enabled, I suggest checking for the appropriate expected value either way.

```
CheckSecurityLevelForServer(
    LOCAL_TCP,
    IsLocalConnectorSecureEnabled() ? TSI_SECURITY_NONE : TSI_PRIVACY_AND_INTEGRITY,
    ep);
```